createtransfers/createtransfers.py

```diff
@@ -194,17 +199,137 @@ def create_variously_encoded_files():
+    zip_path_dirs = os.path.join(HERE, test_transfers, sample_packages,
+                                 var_encoded_dirs)
+    create_variously_encoded_dir_names(zip_path_dirs)
+    create_zip_dance(zip_path_dirs)
+
+    # Create a single zip with one deep directory structure inside using the
+    # application defaults.
+    zip_path_deep = os.path.join(HERE, test_transfers, sample_packages,
+                                 var_deep_zip_transfer)
+    create_deep_transfers(
+        zip_path=zip_path_deep,
+        recursion_depth=createtransfersargs.DEFAULT_DEPTH,
+        number_of_directories=createtransfersargs.DEFAULT_NUMBER_DIRS,
+        number_of_files=createtransfersargs.DEFAULT_NUMBER_FILES)
+    # The first directory above is deliberately made one additional directory
+    # down so that it can be zipped as one. First we have to come back up a
+    # level which we do here.
+    one_deep_zip_path = os.path.join(HERE, test_transfers, sample_packages,
+                                     var_deep_transfer)
+    create_zip_dance(one_deep_zip_path)
+
+    # Create a path where we will create a large zipped transfer folder.
+    zip_path_large = os.path.join(HERE, test_transfers, sample_packages,
+                                  var_large_zip_transfer)
+    # Override the defaults to create a modest 1.1GB zip.
+    create_large_test_transfers(zip_path=zip_path_large,
+                                image_n=35,
+                                video_n=35)
+    # Like the deep transfer, we deliberately drop the large transfers down a
+    # folder level. We come back a level here to zip it all at once.
+    one_large_zip_path = os.path.join(HERE, test_transfers, sample_packages,
+                                      var_large_transfer)
+    create_zip_dance(one_large_zip_path)
+
+
+def create_zip_dance(base_directory):
+    """Perform various manouvers to create zip files from our content and move
+    them into a suitable directory structure.
+    """
+    subdirs = [f for f in os.listdir(base_directory)
+               if os.path.isdir(base_directory)]
+    for dir_ in subdirs:
+        # Create a temporary folder to provide structure inside our zip.
+        sample_package = os.path.join(base_directory, "sample-package")
+        rm_dirs_and_create(sample_package)
+        dir_ = os.path.join(base_directory, dir_)
+        # Move the folder we want to zip underneath the temporary folder.
+        shutil.move(dir_, sample_package)
+        # Re-create the previous folder as we will want to put the final zip
+        # file into here.
+        rm_dirs_and_create(dir_)
+        # Make an archive from our directory.
+        shutil.make_archive(dir_, 'zip', sample_package)
+        # Move the zip file from our temporary folder.
+        move_from = "{}.zip".format(dir_)
+        shutil.move(move_from, dir_)
+        # Finally remove the temporary folder.
+        rm_dirs_and_create(sample_package, True)
```

Comment:
If you don't refactor to create and use a new `rm_dirs` function, then I think at least the kwarg should be explicitly spelled out here, i.e., `rm_dirs_and_create(sample_package, remove_only=True)`.